dotnet/src/Functions/Functions.UnitTests/OpenAPI/Extensions/KernelAIPluginExtensionsTests.cs

```diff
@@ -47,6 +52,39 @@ public KernelAIPluginExtensionsTests()
         this._sut = new OpenApiDocumentParser();
     }
 
+    [Fact]
+    public async Task ItUsesAuthFromOpenAiPluginManifestWhenFetchingOpenApiSpecAsync()
+    {
+        //Arrange
+        var serializerSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new SnakeCaseNamingStrategy()
+            }
+        };
+        using var reader = new StreamReader(ResourcePluginsProvider.LoadFromResource("ai-plugin.json"), Encoding.UTF8);
+        var openAiDocumentContent = JsonConvert.DeserializeObject<JObject>(await reader.ReadToEndAsync(), serializerSettings)!;
+        var actualAuthConfig = openAiDocumentContent["auth"]!.ToObject<OpenAIManifestAuthenticationConfig>(JsonSerializer.Create(serializerSettings))!;
+
+        var openAiDocument = ResourcePluginsProvider.LoadFromResource("ai-plugin.json");
```

Comment:
this unit test is failing because the stream is closed for some reason when the `SendAsync` method is called on the `HttpClient`. still trying to understand why.. any ideas would be appreciated.

Reply:
Look at the test in my alternative solution. I slightly changed it but it works - https://github.com/microsoft/semantic-kernel/blob/db249e5836bb037932cf095aecc42cd92f0469d6/dotnet/src/Functions/Functions.UnitTests/OpenAPI/Extensions/KernelAIPluginExtensionsTests.cs#L55

Reply:
thanks. the test is passing now